web-api/src/models.rs

```diff
@@ -182,26 +182,31 @@ pub(crate) struct DocumentProperty(Value);
 
 impl DocumentProperty {
     pub(crate) fn try_from_value(
-        document_id: &DocumentId,
         property_id: &DocumentPropertyId,
         mut value: Value,
     ) -> Result<Self, InvalidDocumentProperty> {
+        let check_string_property_value = |s: &str| {
+            if s.len() <= 2_048 && GENERIC_STRING_SYNTAX.is_match(s) {
+                Ok(())
+            } else {
+                Err(())
+            }
+        };
+
         match &mut value {
             Value::Bool(_) | Value::Number(_) | Value::Null => {}
             Value::String(string) => {
-                if !is_valid_string_empty_ok(string, 2_048) {
+                let Ok(()) = check_string_property_value(string) else {
                     return Err(InvalidDocumentProperty {
-                        document: document_id.clone(),
                         property: property_id.clone(),
-                        invalid_value: value.clone(),
+                        invalid_value: value,
                         invalid_reason: InvalidDocumentPropertyReason::InvalidString,
                     });
-                }
+                };
```

Comment:
:art: semicolon can be removed if above's suggestion is used
```suggestion
                }
```